### PR TITLE
fix: add a clear button to the experiences

### PR DESCRIPTION
**Pull request by fquellec**
This PR is related to #509
We are adding the ability to delete all files and return to the loading page on every experience. To do this we need to place the file manager and database in the Vuex store, many components have changed their props. This PR needs to be looked at carefully as it impacts a large part of the project. 

Demo: 
image

### CODE REVIEW

**Comment by ffsinger** [comment]
There's a bug in the consent form. Previously we had a BaseButton that generates the zip and triggers a download. The download was done through a hidden BaseButtonDownloadData (thanks to the property `v-show="false"`). Now the button appears on the page. This code hasn't been directly changed, but it may be due to the changes in the BaseButton components.

File: store/index.js
Comment (ffsinger) [inline store/index.js:89]: This function could call the other mutations, I would find it cleaner:
```js
for (const company of config.experiences) {
  this.commit('setSelectedFiles', { key: company, value: [] })
  for (const experience of manifestMap[company]) {
    this.commit('setResult', { company, experience, value: null }
  }
}
this.commit('setCurrentDB', null)
this.commit('setFileManager', null)
```

This could also be turned into an action.

Also, note that currently it clears the results for **all** the experiences. I'm not sure if we want this

File: components/unit/consent-form/UnitConsentForm.vue
Comment (ffsinger) [inline components/unit/consent-form/UnitConsentForm.vue:2]: General note: you can simply use `v-if="fileManager"`

File: store/index.js
Comment (fquellec) [reply store/index.js:89]: We already had one FileManager and one database object shared by all experiments? So there is no need to keep the states of the other experiments, right?

Comment (ffsinger) [reply store/index.js:89]: Ah yes you're right. So we should decide on the direction we wanna take, either:

1. Only keep the state of one experiment. In this case we can simplify `results` and `selectedFiles`.
1. Keep the state of the other experiments. In this case we have to store multiple `fileManager` and `db`

Let's discuss this in the meeting

Comment (ffsinger) [reply store/index.js:89]: As discussed in the meeting we don't need to keep the state of every experiment, and doing so would complicate the code so let's simplify `results` and `selectedFiles`

File: components/unit/files/UnitFiles.vue
Comment (ffsinger) [inline components/unit/files/UnitFiles.vue:203]: This value isn't used by the parent

**Review by ffsinger — APPROVED** [review]
I just moved the consent form to the store and now it's reset after a clear. The consent form code is still very much spaghetti but I'll fix this in a separate PR.

I think we can merge this after the conflicts are fixed

**Pull request merged**
